server/database/models/BudgetRepository.js

```diff
@@ -5,38 +5,42 @@ class BudgetRepository extends AbstractRepository {
     super({ table: "budget" });
   }
 
-  async readAll() {
-    const [rows] = await this.database.query(`SELECT 
-  budget.id,
-  budget.name,
-  budget.amount,
-  DATE_FORMAT(budget.start_date,'%d/%m/%Y') as start_date,
-  DATE_FORMAT(budget.end_date,'%d/%m/%Y') as end_date,
-  category.name AS category_name
-FROM 
-  ${this.table}
-JOIN 
-  category ON budget.category_id = category.id;
-`);
+  async readBudgetsByUser(userId) {
+    const [rows] = await this.database.query(
+      `SELECT 
+        b.id,
+        b.name,
+        b.amount,
+        DATE_FORMAT(b.start_date, '%d/%m/%Y') as start_date,
+        DATE_FORMAT(b.end_date, '%d/%m/%Y') as end_date,
+        c.name AS category_name
+      FROM 
+        ${this.table} b
+      LEFT JOIN 
+        category c ON b.category_id = c.id
+      WHERE 
+        b.user_id = ?`,
+      [userId]
+    );
     return rows;
   }
 
-  async read(id) {
+  async readBudgetById(budget) {
     const [rows] = await this.database.query(
       `SELECT 
-  budget.name,
-  budget.amount,
-  DATE_FORMAT(budget.start_date,'%d/%m/%Y') as start_date,
-  DATE_FORMAT(budget.end_date,'%d/%m/%Y') as end_date,
-  category.name AS category_name
-FROM 
-  ${this.table}
-JOIN 
-  category ON budget.category_id = category.id
-WHERE
-  budget.id = ?;
-`,
-      [id]
+        b.id,
+        b.name,
+        b.amount,
+        DATE_FORMAT(b.start_date, '%d/%m/%Y') as start_date,
+        DATE_FORMAT(b.end_date, '%d/%m/%Y') as end_date,
+        c.name AS category_name
+      FROM 
+        ${this.table} b
+      LEFT JOIN 
+        category c ON b.category_id = c.id
+      WHERE 
+        b.id = ? AND b.user_id = ?`,
+      [budget.id, budget.user_id]
```

Comment:
**nitpick**: You often take an object as a parameter, it's bugging me because I don't see the point of being implicit here. Why don't you take `category_id` & `user_id` as a parameter instead ?

Let's say I want to use your function, looking only at its signature, I cannot use it because I don't know what attribute you expect in your object

Reply:
Let's look at an example:
```javascript
      categoryBelongsToUser = await tables.category.readCategoryById({
        id: req.body.category_id,
        user_id: req.body.user_id,
      });
```
Here you're forced to create another object, even though you have the data

Reply:
I agree with you ! I've refactored most of my parameters.

Though what do you think about keeping add and update models as it is right now ?
In my opinion in this case it's "worth" to use an object. For example this one :
```
async update(budget) {
    const [result] = await this.database.query(
      `UPDATE ${this.table} SET name = ?, amount = ?, start_date = ?, end_date = ?, category_id = ? WHERE id = ? AND user_id = ?`,
      [
        budget.name,
        budget.amount,
        budget.start_date,
        budget.end_date,
        budget.category_id,
        budget.id,
        budget.user_id,
      ]
    );
    return result.affectedRows;
  }
```

Reply:
Agreed, let's keep it like that then :)